fix(files): remove focus outline on file list item click and make the link focusable

ShGKme opened the pull request:
## Summary

* Fix #41490

There is an outline on `focus-visible` state, which appears also on click making a glitch effect.

At the same time, `focus-visible` state was not useful because the link was not focusable.

## Todo

- [x] Show outline only on real focus state
- [x] Make the link focusable by `tabindex` (links usually should not be focusable, but in this case it works close to the button)
- [x] Refactor `linkTo`: replace deprecated `is` attr on `<a>` with a dynamic component:
   - Special attribute `is` is deprecated and removed in Vue 3
   - It is confusing, that `<a>` element is rendered as `<span>` sometimes

Before | After
---|---
**Click** |
![click-before](https://github.com/nextcloud/server/assets/25978914/49e958d2-a505-4d26-b738-e2b26bfb6ab8) | ![click-after](https://github.com/nextcloud/server/assets/25978914/c0672f9e-110f-4159-8681-914c86419b9a)
**Tab** |
![tab-before](https://github.com/nextcloud/server/assets/25978914/d5604d29-78f7-456c-8d83-bd35372dc297) | ![tab-after](https://github.com/nextcloud/server/assets/25978914/51600c53-cc80-4c3a-bddf-04bca9ff8aca)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

ShGKme committed:
fix(files): remove focus outline on file list item click

Signed-off-by: Grigorii K. Shartsev <me@shgk.me>

ShGKme committed:
fix(files): make file list item link focusable

Signed-off-by: Grigorii K. Shartsev <me@shgk.me>

jancborchardt reviewed (approved):
Looks good from the videos, thanks! :)

ShGKme committed:
refactor(files): replace deprecated `is` attr on `<a>` with dynamic component

- Special attribute `is` is deprecated and removed in Vue 3
- It is confusing, that `<a>` element is rendered as `span` sometimes

Signed-off-by: Grigorii K. Shartsev <me@shgk.me>

ShGKme commented on apps/files/src/components/FileEntry/FileEntryName.vue:
```diff
@@ -40,6 +40,7 @@
 	<a v-else
```
It has `href` set by `v-bind="linkTo"`.

But after this comment I found that actually this `<a>` doesn't always render `<a>`

ShGKme commented:
Updates:
1. Set `tabindex` in `linkTo` computed, not on `<a>` in template, because `<a>` is not guaranteed to be rendered as `<a>` here.
2. Refactor `linkTo`: replace deprecated `is` attr on `<a>` with a dynamic component:
   - Special attribute `is` is deprecated and removed in Vue 3
   - It is confusing, that `<a>` element is rendered as `<span>` sometimes

nimishavijay reviewed (approved):
Nice! Looks great design-wise!

ShGKme committed:
chore(assets): Recompile assets

Signed-off-by: Grigorii K. Shartsev <me@shgk.me>

ShGKme commented:
Rebased onto master.

ShGKme commented:
/backport to stable28

ShGKme commented:
/backport 2f18651 fedd531 e6bdbdb to stable28
